pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -58,11 +65,54 @@ Examples:
 				newBuild = buildutil.GenerateBuildFromBuild(build)
 			}
 
+			var wg sync.WaitGroup
+			redirect := make(chan string, 1)
+			if follow {
+				set := labels.Set(newBuild.Labels)
+				selector := labels.SelectorFromSet(set)
+				// Add a watcher for the build about to start
+				watcher, err := client.Builds(namespace).Watch(selector, labels.Everything(), cmdutil.GetFlagString(cmd, "api-version"))
+				checkErr(err)
+
+				wg.Add(1)
+				go func(watcher watch.Interface, redirect <-chan string) {
+					defer watcher.Stop()
+					defer wg.Done()
+
+					// Wait for the build name; w/o it there is not
+					// much that can happen here
+					buildName := <-redirect
+
+					for event := range watcher.ResultChan() {
+						build := event.Object.(*buildapi.Build)
+						// Iterate over watcher's results and search for
+						// the build we just started. Also make sure that
+						// the build is running
+						if build.Name == buildName && build.Status == buildapi.BuildStatusRunning {
+							rd, err := client.Get().Namespace(namespace).Prefix("redirect").Resource("buildLogs").Name(buildName).Stream()
+							checkErr(err)
+							defer rd.Close()
+
+							_, err = io.Copy(out, rd)
+							checkErr(err)
+							return
+						}
+					}
+				}(watcher, redirect)
+			}
+
+			// Start a build
 			newBuild, err = client.Builds(namespace).Create(newBuild)
```

Comment:
why can't you just do this before you start the follow logic, rather than using a channel to send newBuild.Name in later?